base/singleton.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "base/macros.hpp"
+
+namespace base
+{
+// Simple singleton class.
+template <typename T>
+class Singleton
```

Comment:
This is not a Singleton, because the method Instance returns T, not Singleton. Moreover, it's possible to create instances of T, just by calling their constructor. So, I don't see why do we need this class.